compilator/compilator_test.go

```diff
@@ -60,9 +65,18 @@ func TestCompilationEmpty(t *testing.T) {
 	}()
 
 	<-waitCh
+
+	contents, err := ioutil.ReadFile(metrics)
+	assert.Nil(err)
+	assert.Regexp(regexp.MustCompile(expected), string(contents))
 }
 
 func TestCompilationBasic(t *testing.T) {
+	metrics := "test-metrics.csv"
+	expected := `.*,compilator,start\n.*,compilator::job::test-release/ruby-2.5,start\n.*,compilator::job::wait::test-release/ruby-2.5,start\n.*,compilator::job::wait::test-release/ruby-2.5,done\n.*,compilator::job::run::test-release/ruby-2.5,start\n.*,compilator::job::run::test-release/ruby-2.5,done\n.*,compilator::job::test-release/ruby-2.5,done\n.*,compilator::job::test-release/go-1.4,start\n.*,compilator::job::wait::test-release/go-1.4,start\n.*,compilator::job::wait::test-release/go-1.4,done\n.*,compilator::job::run::test-release/go-1.4,start\n.*,compilator::job::run::test-release/go-1.4,done\n.*,compilator::job::test-release/go-1.4,done\n.*,compilator::job::test-release/consul,start\n.*,compilator::job::wait::test-release/consul,start\n.*,compilator::job::wait::test-release/consul,done\n.*,compilator::job::run::test-release/consul,start\n.*,compilator::job::run::test-release/consul,done\n.*,compilator::job::test-release/consul,done
+.*/compilator.go:.*,compilator,done`
```

Comment:
Really not a fan of the giant line. Thinking of something more like
```go
expected := []string{
	`.*,compilator,start`,
	…
}
actual := strings.Split(strings.TrimSpace(string(contents)), "\n")
if assert.Len(actual, len(expected)) {
	for lineno, expectedLine := range expected {
		assert.Regexp(regexp.MustCompile(expectedLine), actual[lineno])
	}
}
```
(Though it looks like if we match line-by-line we can use `string.HasSuffix` instead of doing regexps)

Also, is line 77 supposed to be off by itself like that?